### PR TITLE
Add deeplink schemes

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR adds deeplinks support for app with authentication route.

### Result and observation

Kapture 2019-04-19 at 12 49 37

To emulate this, you can read this guide: https://github.com/CoorpAcademy/mobile/tree/da7e566bc2b63e3eb6a3f51537f7f040c337426a#deeplinks

The route is: `coorpacademyapp://authentication/<YOUR_TOKEN>`

- [ ] **Breaking changes ?**
- [ ] **Extra dependency ?**

### Testing Strategy

- [ ] Already covered by tests
- [x] Manual testing
- [ ] Unit testing

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,44 +1,50 @@
...
+      android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
+      android:screenOrientation="behind"
+      android:windowSoftInputMode="adjustResize"
+      android:exported="true"
+      android:launchMode="singleTask">
+      <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="coorpacademyapp" />
```
Comment (VincentCATILLON) [inline android/app/src/main/AndroidManifest.xml]: This is what we need to set the scheme used by the app (Android)

File: ios/Coorpacademy/AppDelegate.m [hunk 2]
```diff
@@ -11,11 +11,29 @@
 #import <React/RCTRootView.h>
 #import "RNSplashScreen.h"
 #import "Orientation.h"
+#import <React/RCTLinkingManager.h>
 
 @import Firebase;
 
 @implementation AppDelegate
 
+- (BOOL)application:(UIApplication *)application
```
Comment (VincentCATILLON) [inline ios/Coorpacademy/AppDelegate.m]: These methods are used to catch deep and universal links

File: ios/Coorpacademy/Info.plist [hunk 3]
```diff
@@ -2,6 +2,15 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>CFBundleURLTypes</key>
+	<array>
+		<dict>
+			<key>CFBundleURLSchemes</key>
+			<array>
+				<string>coorpacademyapp</string>
```
Comment (VincentCATILLON) [inline ios/Coorpacademy/Info.plist]: This is what we need to set the scheme used by the app (iOS)

File: src/navigator/index.js [hunk 4]
```diff
@@ -26,13 +25,6 @@ import navigationOptions, {
 
 const appNavigator = createStackNavigator(
   {
-    Splash: {
-      screen: SplashScreen,
-      navigationOptions: {
-        ...navigationOptionsWithoutHeader,
-        gesturesEnabled: false
-      }
-    },
```
Comment (VincentCATILLON) [inline src/navigator/index.js]: This screen is not useful anymore.

**Comment by VincentCATILLON** [comment]
Waiting for @Fausto95 review

**Comment by Fausto95** [comment]
It's ok for the auth && splash screen

**Pull request merged**